src/plugins/proposal.ts

```diff
@@ -0,0 +1,458 @@
+    active?: boolean
+    boosted?: boolean
+    dao?: Address
+    expiresInQueueAt?: Date
+    expiresInQueueAt_gte?: Date
+    expiresInQueueAt_lte?: Date
+    expiresInQueueAt_gt?: Date
+    executedAfter?: Date
+    executedBefore?: Date
+    id?: string
+    proposer?: Address
+    proposalId?: string
+    stage?: IProposalStage
+    stage_in?: IProposalStage[]
+    plugin?: Address
+    orderBy?: ProposalQuerySortOptions
+    type?: ProposalTypeNames
+    [key: string]: any | undefined
+  }
+}
+
+export interface IProposalBaseCreateOptions {
+  dao: Address
+  description?: string
+  descriptionHash?: string
+  title?: string
+  tags?: string[]
+  plugin?: Address
+  url?: string
+  proposalType?: ProposalTypeNames
+}
+
+export interface IProposalState {
+  // TODO: semantically order props
+  id: string
+  dao: IEntityRef<DAO>
+  votingMachine: Address
+  plugin: IEntityRef<Plugin>
+  closingAt: Number
+  createdAt: Number | Date
+  descriptionHash?: string
+  description?: string
+  name: string,
+  executedAt: Number
+  organizationId: string
+  paramsHash: string
+  proposal: IEntityRef<Proposal>
+  proposer: Address
+  resolvedAt: Number
+  tags?: string[]
+  title?: string
+  totalRepWhenCreated: BN
+  totalRepWhenExecuted: BN
+  type: ProposalTypeNames
+  url?: string
+  votesFor: BN
+  votesAgainst: BN
+  votesCount: number
+  voteOnBehalf: Address
+  winningOutcome: IProposalOutcome
+
+  // Genesis Protocol Proposal
+  queue: IEntityRef<Queue>
+  quietEndingPeriodBeganAt: Number
+  stage: IProposalStage
+  accountsWithUnclaimedRewards: Address[]
+  boostedAt: Number
+  upstakeNeededToPreBoost: BN
+  stakesFor: BN
+  stakesAgainst: BN
+  preBoostedAt: Number
+  genesisProtocolParams: IGenesisProtocolParams
+  executionState: IExecutionState
+  expiresInQueueAt: Number
+  downStakeNeededToQueue: BN
+  confidenceThreshold: number
+}
+
+export abstract class Proposal extends Entity<IProposalState> {
+
+  public static fragment: { name: string, fragment: DocumentNode } | undefined
+
+  // TODO: dynamically generate these from Proposals
+  public static baseFragment: DocumentNode = gql`fragment ProposalFields on Proposal {
+      id
+      accountsWithUnclaimedRewards
+      boostedAt
+      closingAt
+      confidenceThreshold
+      createdAt
+      dao {
+        id
+        schemes {
+          id
+          address
+        }
+      }
+      description
+      descriptionHash
+      executedAt
+      executionState
+      expiresInQueueAt
+      genesisProtocolParams {
+        id
+        activationTime
+        boostedVotePeriodLimit
+        daoBountyConst
+        limitExponentValue
+        minimumDaoBounty
+        preBoostedVotePeriodLimit
+        proposingRepReward
+        queuedVotePeriodLimit
+        queuedVoteRequiredPercentage
+        quietEndingPeriod
+        thresholdConst
+        votersReputationLossRatio
+      }
+      gpRewards {
+        id
+      }
+      scheme {
+        ...PluginFields
```

Comment:
This doesn't look to be defined below along with the Proposal fragments.

Reply:
Yeah, it was because that fragment has another inside elsewhere. I fixed it because it was a bit confusing